src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  let parts = sourceString.split(';');
+
+  parts = parts.filter((part) => part.trim() !== '');
+
+  for (const part of parts) {
+    const partKeyValue = part.split(':');
+
+    for (let i = 0; i < partKeyValue.length; i++) {
+      const key = partKeyValue[0].trim();
+      const value = partKeyValue[1].trim();
+
+      result[key] = value;
+    }
+  }
```

Comment:
The use of a 'for...of' loop to iterate over 'parts' is acceptable, but the nested 'for' loop is unnecessary and does not function correctly. It will repeatedly overwrite the same keys and values. You should remove the nested 'for' loop and directly assign the key-value pairs inside the 'for...of' loop.